geode-core/src/distributedTest/java/org/apache/geode/internal/cache/wan/asyncqueue/AsyncEventQueuePausedDUnitTest.java

```diff
@@ -0,0 +1,242 @@
+  private static ClientVM client;
+
+  @Before
+  public void beforeClass() throws Exception {
+    locator = lsRule.startLocatorVM(0);
+    server1 = lsRule.startServerVM(1, "group1", locator.getPort());
+    server2 = lsRule.startServerVM(2, "group1", locator.getPort());
+    int serverPort = server1.getPort();
+    client =
+        lsRule.startClientVM(3, new Properties(),
+            clientCacheFactory -> configureClientCacheFactory(clientCacheFactory, serverPort));
+  }
+
+  private static void configureClientCacheFactory(ClientCacheFactory ccf, int... serverPorts) {
+    for (int serverPort : serverPorts) {
+      ccf.addPoolServer("localhost", serverPort);
+    }
+    ccf.setPoolReadTimeout(10 * 60 * 1000); // 10 min
+    ccf.setPoolSubscriptionEnabled(true);
+  }
+
+  @Test
+  public void whenAEQCreatedInPausedStateThenListenersMustNotBeInvoked() {
+    final AEQandRegionProperties props = new AEQandRegionProperties(regionShortcut, isParallel);
+    server1.invoke(() -> {
+      createRegionAndDispatchingPausedAEQ(props);
+    });
+    server2.invoke(() -> {
+      createRegionAndDispatchingPausedAEQ(props);
+    });
+    client.invoke(() -> {
+      createClientRegion();
+    });
+
+    server1.invoke(() -> {
+      validateAEQDispatchingIsPaused();
+    });
+
+    server2.invoke(() -> {
+      validateAEQDispatchingIsPaused();
+    });
+
+    // Resume dispatching.
+    server1.invoke(() -> {
+      ClusterStartupRule.getCache().getAsyncEventQueue("aeqID").resumeEventDispatching();
+    });
+
+    server2.invoke(() -> {
+      ClusterStartupRule.getCache().getAsyncEventQueue("aeqID").resumeEventDispatching();
+    });
+
+    // Validate dispatching resumed.
+    await().atMost(1, TimeUnit.MINUTES).until(() -> {
+
+      final int count1 = server1.invoke(() -> getEventDispatchedSize());
+      final int count2 = server2.invoke(() -> getEventDispatchedSize());
+      if ((count1 + count2) == 1000) {
+        return true;
+      } else {
+        return false;
+      }
+    });
+
+  }
+
+  @NotNull
+  private static Integer getEventDispatchedSize() {
+    Cache cache = ClusterStartupRule.getCache();
+    AsyncEventQueueImpl aeq = (AsyncEventQueueImpl) cache.getAsyncEventQueue("aeqID");
+    MyAsyncEventListener listener = (MyAsyncEventListener) aeq.getAsyncEventListener();
+    return listener.getEventsMap().size();
+  }
+
+  private static void createClientRegion() {
+    ClientCache cache = ClusterStartupRule.getClientCache();
+    Region region = cache.createClientRegionFactory(ClientRegionShortcut.PROXY)
+        .create("region");
+    for (int i = 0; i < 1000; i++) {
+      region.put(i, i);
+    }
+  }
+
+  private static void createRegionAndDispatchingPausedAEQ(AEQandRegionProperties props) {
+    Cache cache = ClusterStartupRule.getCache();
+    cache.createAsyncEventQueueFactory()
+        .pauseEventDispatching()
+        .setParallel(props.isParallel())
+        .setPersistent(isPersistent(props))
```

Comment:
How about setting batch size to 1? Does it help here...

Reply:
There were 1000 puts thats why I imagined that the default will be sufficient.